src/zarr.v2.cpp

```diff
@@ -22,6 +24,21 @@ compressed_zarr_v2_init()
     }
     return nullptr;
 }
+
+std::string
+sample_type_to_dtype(SampleType t)
+
+{
+    const std::string dtype_prefix =
+      std::endian::native == std::endian::big ? ">" : "<";
+
+    std::string table[] = { "u1", "u2", "i1", "i2", "f4", "u2", "u2", "u2" };
```

Comment:
It seems that the size of the table is based on the assumption that `SampleType` enum values directly correspond to indices in the table array. If that’s the case, this type of logic is very fragile. I would use a switch statement that covers all the cases of `SampleType`.

```cpp
std::string sample_type_to_dtype(SampleType t) {
    const auto prefix = std::endian::native == std::endian::big ? ">" : "<";

    switch (t) {
        case SampleType_u8:
            return prefix + "u1";
        case SampleType_u16:
            return prefix + "u2";
        case SampleType_i8:
            return prefix + "i1";
        case SampleType_i16:
            return prefix + "i2";
        case SampleType_f32:
            return prefix + "f4";
        case SampleType_u10:
        case SampleType_u12:
        case SampleType_u14:
            return prefix + "u2";
        default:
            throw std::runtime_error("Invalid SampleType: " + std::to_string(static_cast<int>(t)));
    }
}
```

1. The switch statements clearly express intent of handling different cases based on the enum value.
2. Switch statements are often checked at compile time to ensure all enum values are covered.
3. Adding new cases or modifying existing ones will be straightforward.
4. Switch typically compile to jump tables or direct comparisons which may result in better performance.

I think this applies to `std::string sample_type_to_dtype(SampleType t)` as well.

Reply:
I think this is fair.